### PR TITLE
Prepare for codegen

### PR DESCRIPTION
**Pull request by ChenZhao44**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -4,23 +4,20 @@ authors = ["Roger-Luo <rogerluo.rl18@gmail.com> and contributors"]
 version = "0.1.1"
```
Comment (Roger-luo) [inline Project.toml:4]: compact is still necessary, you need to update the compat version

File: src/BQCESubroutine.jl [hunk 2]
```diff
@@ -1,37 +1,29 @@
 module BQCESubroutine
 
-export broutine!, @broutine
+export broutine!, @broutine, log2dim, log2i
 
-using MLStyle
-using ExprTools
+using MLStyle, Expronicon, ExprTools
```
Comment (Roger-luo) [inline src/BQCESubroutine.jl]: don't using multiple modules in one line, this is a style we use in this org.

**Review by Roger-luo — COMMENTED** [review]
the test has not passed, we need at least the subspace_mul functions to pass test, have you checked that?

**Comment by ChenZhao44** [comment]
> the test has not passed, we need at least the subspace_mul functions to pass test, have you checked that?

The test failed because of StaticInt. I'm not sure I fixed it in the right way. But it works for me at least.

**Pull request merged**